nncf/torch/model_transformer.py

```diff
@@ -0,0 +1,97 @@
+from nncf.torch.nncf_network import PTInsertionPoint
+
+
+class PTModelTransformer(ModelTransformer):
+    def __init__(self, model: NNCFNetwork):
+        super().__init__(model)
+        self._node_to_op_address_mapping = model.get_node_to_op_address_mapping()
+
+    def transform(self, transformation_layout: PTTransformationLayout) -> NNCFNetwork:
+        transformations = transformation_layout.transformations
+        insertion_transformations = []
+        extraction_transformations = []
+        bias_correction_transformations = []
+        for transformation in transformations:
+            if isinstance(transformation, PTInsertionCommand):
+                insertion_transformations.append(transformation)
+            if isinstance(transformation, PTModelExtractionCommand):
+                extraction_transformations.append(transformation)
+            if isinstance(transformation, PTBiasCorrectionCommand):
+                bias_correction_transformations.append(transformation)
+
+        if extraction_transformations:
+            return self._apply_extraction_transformations(extraction_transformations)
+
+        if insertion_transformations:
+            self._apply_output_insertion_transformations(insertion_transformations)
+        if bias_correction_transformations:
+            self._apply_bias_correction_transformations(bias_correction_transformations)
+        return self._model
+
+    def _apply_output_insertion_transformations(self, transformations: List[TransformationCommand]) -> None:
+        fns_grouped_by_points = {}  # type: Dict[PTInsertionPoint, List[Tuple[Callable, TransformationPriority]]]
+        for transformation_command in transformations:  # type: PTInsertionCommand
+            target_point = transformation_command.target_point  # type: PTTargetPoint
+            target_node_name = target_point.target_node_name
+            pt_ip = PTInsertionPoint(
+                target_type=target_point.target_type,
+                op_address=self._node_to_op_address_mapping[target_node_name],
+                input_port_id=target_point.input_port_id,
+            )
+            fn = transformation_command.fn
+            if target_point.type is TargetType.OPERATION_WITH_WEIGHTS:
+                fn = UpdateWeight(fn)
+            tup = (fn, transformation_command.priority)
+            if pt_ip not in fns_grouped_by_points:
+                fns_grouped_by_points[pt_ip] = [tup]
+            else:
+                fns_grouped_by_points[pt_ip].append(tup)
+
+        for pt_ip, fn_list_with_priority in fns_grouped_by_points.items():
+            fn_list_with_priority = sorted(fn_list_with_priority, key=lambda x: x[1])
+            self._model.insert_at_point(pt_ip, [x[0] for x in fn_list_with_priority])
+
+    def _apply_extraction_transformations(self, transformations: List[TransformationCommand]) -> nn.Module:
+        assert len(transformations) == 1, "Support only one node extraction"
```

Comment:
Please, see the other backends to avoid such assert.

Reply:
ok, changed 
But now, if somebody will call several extraction_transformations will apply only last.